yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/SimpleAgeVerification.java

```diff
@@ -0,0 +1,47 @@
+package com.yoti.api.client.spi.remote;
+
+import static java.lang.Boolean.parseBoolean;
+
+import com.yoti.api.client.AgeVerification;
+import com.yoti.api.client.Attribute;
+import com.yoti.api.client.spi.remote.util.Validation;
+
+public class SimpleAgeVerification implements AgeVerification {
+
+    private final Attribute<String> derivedAttribute;
+    private final int ageVerified;
+    private final String checkPerformed;
+    private final boolean result;
+
+    public SimpleAgeVerification(Attribute<String> derivedAttribute) {
+        Validation.notNull(derivedAttribute, "derivedAttribute");
+        Validation.matchesPattern(derivedAttribute.getName(), "[^:]+:(?!.*:)[0-9]+", "attribute.name");
```

Comment:
I would have moved this process into a Processor and make this class as stupid as possible.
It should just get the data from the constructor, set them and return the AgeVerification object.

Reply:
Here, the _SimpleAgeVerification_ is just asserting that its being created with the correct information.  It's validating it's own creation / being responsible for (what will be) it's own content, which is reasonable.

It could go into a _SimpleAgeVerification_ factory, but I don't think there's enough going on here to justify creating one.  We're only creating one type of object, and we're not going to reuse a factory anywhere else.

And it's happening inside the constructor, so there's not really a sensible way to inject some other component.  You might use static code, but I'm trying to avoid that in most cases because it often gets tricky to unit test cleanly.

Reply:
Ok